flink-runtime-web/web-dashboard/src/app/pages/job/timeline/job-timeline.component.ts

```diff
@@ -143,16 +189,18 @@ export class JobTimelineComponent implements AfterViewInit, OnDestroy {
     this.mainChartInstance.tooltip({
       title: 'name'
     });
-    this.mainChartInstance.on('click', (e: SafeAny) => {
+    this.mainChartInstance.on('click', (e: { x: number; y: number }) => {
       if (this.mainChartInstance.getSnapRecords(e).length) {
-        const data = (this.mainChartInstance.getSnapRecords(e)[0] as SafeAny)._origin;
+        const data = ((this.mainChartInstance.getSnapRecords(e)[0] as unknown) as {
```

Comment:
... can we really not avoid typing this as `number[0]` and then later on typecasting it to access `._origin` after all?

Reply:
`… as number as { _origin: }` will not work, no. The whole reason `as unknown` is in here in the first place is because TS will not let you type-assert a known type into another one when there is no overlap and it could never work. It then recommends the only way around that, which is asserting it to `unknown` first.

Really this is all rather "broken", obviously the types do not match what is going on here. But I haven't (yet) addressed that.

Reply:
Sorry, my message was a bit garbled. What I meant to point out was that it's rather strange that we apparently have a `number[]`, but then suddenly access property `._origin`. But you already addressed that in your answer